### PR TITLE
Add integration tests lm generate torch tf

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Add integration tests for all LM models that are able to generate language.

- All integration tests use `do_sample=False` (greedy) generation and verify that TF 2.0 and PT yield the same results. 
- Fixed a small bug with TFXLMModelWithLMHead

### CODE REVIEW

File: w! [hunk 1]
```diff
@@ -0,0 +1,430 @@
+# coding=utf-8
```
Comment (julien-c) [inline w!]: extraneous file?

**Comment by julien-c** [comment]
Approx. how long do these tests take on a single V100?

File: w! (hunk 1 above)
Comment (patrickvonplaten) [reply w!]: sorry - have to be a bit careful with my vim command!

**Comment by patrickvonplaten** [comment]
> Approx. how long do these tests take on a single V100?

Don't know how long they take on a V100. On a cpu, all tests combined (7 model tests for PT and 6 model tests for TF) take less than 10min (whereas `test_modeling_tf_xlnet.py`, `test_modeling_xlnet.py` and `test_modeling_transfo_xl.py` combined take ca. 8min)

**Review by julien-c — APPROVED** [review]

File: src/transformers/modeling_tf_xlm.py [hunk 2]
```diff
@@ -408,7 +408,7 @@ def call(
             inputs_embeds = self.embeddings(input_ids)
 
         tensor = inputs_embeds + self.position_embeddings(position_ids)
-        if langs is not None and self.use_lang_emb:
+        if langs is not None and self.use_lang_emb and self.n_langs > 1:
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_xlm.py]: Nice catch!

**Review by LysandreJik — APPROVED** [review]
Nice, LGTM!

**Comment by patrickvonplaten** [comment]
Good to merge for me

**Pull request merged**